test/integration/dai.spec.js

```diff
@@ -135,50 +136,97 @@ test(
   300000
 );
 
-test(
-  'can sell Dai',
-  async () => {
-    let tx, error;
-
-    if (process.env.NETWORK === 'test') {
-      await createDaiAndPlaceLimitOrder(exchange);
-    }
-
-    try {
-      tx = await exchange.sellDai('0.1', WETH);
-    } catch (err) {
-      console.error(err);
-      error = err;
-    }
-
-    expect(tx).toBeDefined();
-    expect(error).toBeUndefined();
-  },
-  600000
-);
+xdescribe('OasisExchangeService', () => {
+  test(
+    'can sell Dai',
+    async () => {
+      let tx, error;
+
+      if (process.env.NETWORK === 'test') {
+        await createDaiAndPlaceLimitOrder(exchange);
+      }
+
+      try {
+        tx = await exchange.sellDai('0.1', WETH);
+      } catch (err) {
+        console.error(err);
+        error = err;
+      }
+
+      expect(tx).toBeDefined();
+      expect(error).toBeUndefined();
+    },
+    600000
+  );
+
+  test(
+    'can buy Dai',
+    async () => {
+      let tx, error;
+      await checkWethBalance();
+
+      if (process.env.NETWORK === 'test') {
+        await createDaiAndPlaceLimitOrder(exchange, true);
+      }
+
+      try {
+        tx = await exchange.buyDai('0.1', WETH);
+      } catch (err) {
+        console.error(err);
+        error = err;
+      }
+
+      expect(tx).toBeDefined();
+      expect(error).toBeUndefined();
+    },
+    600000
+  );
+});
 
-test(
-  'can buy Dai',
-  async () => {
-    let tx, error;
-    await checkWethBalance();
-
-    if (process.env.NETWORK === 'test') {
-      await createDaiAndPlaceLimitOrder(exchange, true);
-    }
-
-    try {
-      tx = await exchange.buyDai('0.1', WETH);
-    } catch (err) {
-      console.error(err);
-      error = err;
-    }
-
-    expect(tx).toBeDefined();
-    expect(error).toBeUndefined();
-  },
-  600000
-);
+// Running these consecutively seems to confuse jest
+// and causes unexpected errors. Each will work if
+// run independently from the others
```

Comment:
can you say more about what you're experiencing here? would it be more reliable to separate this into different test files or something?

Reply:
@levity I'm glad you brought this up (I actually mentioned it yesterday in the sprint planning meeting too). When I was originally building the integration tests for CDP functionality and the original Oasis service, I experienced problems similar to these. I think we may have even paired once on them a long time ago?

The only definitive thing I can say really is that jest acts very unpredictably with mainnet. I've seen all kinds of weird things happen when running these. I can elaborate on what they look like and patterns I've noticed, but in general I think it would be really worthwhile to figure out a more sustainable way to automate mainnet testing.